build.gradle

```diff
@@ -3,26 +3,329 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.2.3'
-        classpath 'de.undercouch:gradle-download-task:1.0'
-        classpath 'com.neenbedankt.gradle.plugins:android-apt:1.4'
+        classpath 'org.jfrog.buildinfo:build-info-extractor-gradle:3.1.1'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.2'
-        classpath 'com.jakewharton.sdkmanager:gradle-plugin:0.12.0'
-        classpath 'com.novoda:gradle-android-command-plugin:1.3.0'
-        classpath 'com.github.skhatri:gradle-s3-plugin:1.0.2'
+        classpath 'ch.netzwerg:gradle-release-plugin:1.1.0'
     }
 }
 
-allprojects {
-  apply plugin: 'android-sdk-manager'
+apply plugin: 'maven-publish'
+apply plugin: 'com.jfrog.artifactory'
+apply plugin: 'com.jfrog.bintray'
+apply plugin: 'ch.netzwerg.release'
 
-  group = 'io.realm'
-  version = file("${rootDir}/version.txt").text.trim();
-  repositories {
-    jcenter()
-  }
+def currentVersion = new File("${projectDir}/version.txt").text.trim();
+
+task checkEnvironment(group: 'check', description: 'Check the user provided gradle.properties') << {
+    def ndkDir = System.env.NDK_HOME
+    if (!ndkDir) {
+        throw new GradleException('The NDK_HOME environment variable is not set.')
+    } else if (ndkDir.endsWith('/')) {
+        throw new GradleException('The NDK_HOME environment variable ends with /')
```

Comment:
That mainly comes from the fact that it's an environment variable, which normally does not include trailing slashes for folders. But I can add the removal just for good measure. But it will need to be cross platform, so removing forward slashes as well